typescript/ws-protocol-examples/src/examples/mcap-record.ts

```diff
@@ -190,13 +197,25 @@ async function main(address: string, options: { output: string }): Promise<void>
           log("received message for unknown subscription %s", event.subscriptionId);
           return;
         }
-        void writer.addMessage({
-          channelId: subscription.mcapChannelId,
-          sequence: subscription.messageCount++,
-          logTime: BigInt(Date.now()) * 1_000_000n,
-          publishTime: event.timestamp,
-          data: new Uint8Array(event.data.buffer, event.data.byteOffset, event.data.byteLength),
-        });
+
+        // If we can't acquire the lock, we will drop the message to avoid potential OOM issues.
```

Comment:
Maybe we could have a setting for this? It could be useful in some situations to record everything even if it takes longer than realtime. We should make sure that after ^C the writer still gets closed cleanly — ideally finishing everything that was in the write queue already

Reply:
done